### PR TITLE
[F.Lute] add container support

### LINKED ISSUE #1: Auto-deployments via Heroku
- [x] Heroku.yml stack
- [x] support environment variables
- [x] Dockerfile

### PR DESCRIPTION
**Pull request by sr229**
Closes #1

### CODE REVIEW

File: bot/structures/config.py [hunk 1]
```diff
@@ -15,3 +16,12 @@ class Config:
     def from_file(cls, file_name: str = DEFAULT_PATH) -> 'Config':
         with open(file_name) as f:
             return cls(**json.load(f))
+    @classmethod
```
Comment (Martmists-GH) [inline bot/structures/config.py:20]: add a newline before this

File: main.py [hunk 2]
```diff
@@ -1,8 +1,13 @@
 from bot.bot import MusicBot
 from bot.structures.config import Config
+import os
 
 if __name__ == "__main__":
-    cfg = Config.from_file("config.json")
+    cfg = Config.from_file("./config.json")
```
Comment (Martmists-GH) [inline main.py]: Maybe check if an env var exists instead and select the config init based on that:

```py
if os.environment.get("FLUTE_PREFIX") is not None:
    # get from env
else:
    # get from file
```

File: app.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "Entertainment",
+      "python",
+      "bot"
+    ],
+    "repository": "https://github.com/ClarityCafe/F.Lute",
+    "logo": "https://cdn.discordapp.com/avatars/436026947974463488/7eb55b489b3335e1bf78891bc3b38ae1.png?size=512",
+    "env": {
+      "FLUTE_PREFIX": {
+        "description": "Prefix for the bot",
+        "generator": "flute "
```
Comment (Martmists-GH) [inline app.json:14]: Prefix is a list, not a string. See the config.example for the default value.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
+FROM python:slim
+
+# Make container root
+USER root
+RUN mkdir /app
+
+# Install Python dependencies
+COPY requirements.txt /app
```
Comment (Martmists-GH) [inline Dockerfile:8]: For this to work, requirements.txt must be present, we will mostly be using these packages: 
```
discord.py
essentia
numpy
```

**Review by Martmists-GH — CHANGES_REQUESTED** [review]

File: app.json (hunk 3 above)
Comment (sr229) [reply app.json:14]: I don't think we can do lists in environment variables.

Comment (Martmists-GH) [reply app.json:14]: Just split the string by comma or something in the python code.

File: main.py (hunk 2 above)
Comment (sr229) [reply main.py]: We'll just check if file exists for now.

File: main.py [hunk 5]
```diff
@@ -5,8 +5,9 @@
 if __name__ == "__main__":
     cfg = Config.from_file("./config.json")
```
Comment (Martmists-GH) [inline main.py]: Might want to remove this line since it's overwritten afterwards.

**Comment by Martmists-GH** [comment]
LGTM

**Pull request merged**